pkg/operator/ceph/cluster/osd/osd.go

```diff
@@ -221,6 +222,15 @@ func (c *Cluster) Start() error {
 		return errors.Wrapf(err, "failed to replace an OSD that needs migration to new backend store in namespace %q", namespace)
 	}
 
+	// identify OSDs within the fabric fault domain that need to be transferred to a new node
```

Comment:
Why do we need this portion? The source host is already deleted before here, isn't it?

Reply:
Plus, we must avoid making any changes on the original rook.

Reply:
> In the `nvmeofstorage` controller, we only updated the `cluster CR` and marked it as needing to be transferred to new host.
> So source host deletion happened here. This approach was taken to sequentially handle multiple OSD compute node failures via rook-operator.

I don't follow this, can you clarify what you explained? Why this codes should be added here instead of nvmeofstroage controller.